modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TrafficSources.js

```diff
@@ -120,7 +120,10 @@ export default function TrafficSources({
 							{trafficSources.map((entry) => {
 								const hasDetails =
 									entry?.countryKeywords !== undefined ||
-									entry?.details !== undefined;
+									(entry?.referringPages !== undefined &&
+										entry?.referringDomains !==
+											undefined) ||
+									entry?.referringSocialMedia !== undefined;
```

Comment:
Just wondering, could all of these `!== undefined` be elided?

```js
const hasDetails =
	entry?.countryKeywords ||
	(entry?.referringPages && entry?.referringDomains) ||
	entry?.referringSocialMedia;
```

Because without looking at what the types are, it looks like they might be "`undefined` or truthy".

Reply:
These keys may no exists in the object that I received. countryKeywords is the "details" for Organic and Paid, referringPages and referringDomains for Referral traffic and referringSocial media for Social traffic. This check could be done without "!== undefined", I suppose. But if everything is undefined, hasDetails is undefined, not false. Is that right?

Reply:
[Falsy values in JS](https://developer.mozilla.org/en-US/docs/Glossary/Falsy): `false`, `null`, `undefined`, `0`, `''` (also some edgier ones listed on that page, like `-0`, `NaN`, and `0n`).

If you want `hasDetails` to be boolean at the end you can put a `!!` in the front of the expression, but you often don't need to, because the code that you're passing too usually only cares about falsiness/truthiness, not literal `false`/`true`. This is [the case for JSX](https://react-cn.github.io/react/tips/false-in-jsx.html), for example, like you have here; that is, when you interpolate `{false}` or `{undefined}`, React is going to [do the right thing](https://reactjs.org/docs/jsx-in-depth.html#booleans-null-and-undefined-are-ignored).

Up to you though. I personally, find the repeated `!== undefined` to be clutter.

Reply:
I'll annotate those articles to read 👍 Thanks for the feedback. I'm going to change it and remove those `!== undefined`.